src/default-xpring-client.ts

```diff
@@ -299,7 +300,13 @@ class DefaultXpringClient implements XpringClientDecorator {
       await this.getBalance(address)
       return true
     } catch (e) {
-      return false
+      if (
+        Object.prototype.hasOwnProperty.call(e, 'code') &&
```

Comment:
I'm surprised you can't use typescripts optional chaining here:
```
if e.code? === grpc.status.NOT_FOUND
```

If code is `undefined` we'll safely unwrap it to undefined and the check will fail. Otherwise, it'll be unwrapped to a status code and compared.

Reply:
So I'm pretty sure we actually can use this construct.  I'll double-check with typescript people here to make sure nothing stupid happens, but if they approve I'll change that.

Reply:
so the optional chaining syntax doesn't seem to work in that particular way, but I updated the if statement so it's a bit shorter.